applications/CompressiblePotentialFlowApplication/python_scripts/stochastic_potential_flow_response.py

```diff
@@ -0,0 +1,344 @@
+        XMCAlgorithmInputDictionary["hierarchyOptimiser"] = hierarchyCostOptimiser
+        XMCAlgorithmInputDictionary["stoppingCriterion"] = criterion
+
+        self.xmc_analysis = xmc.XMCAlgorithm(**XMCAlgorithmInputDictionary)
+
+        if (parameters["solverWrapperInputDictionary"]["asynchronous"] is True):
+            self.xmc_analysis.runAsynchronousXMC()
+        else:
+            self.xmc_analysis.runXMC()
+
+    def _GetAdjointParameters(self):
+        with open(self.response_settings["adjoint_settings"].GetString(),'r') as parameter_file:
+            adjoint_parameters = Parameters( parameter_file.read() )
+
+        return adjoint_parameters
+
+class SimulationScenario(potential_flow_analysis.PotentialFlowAnalysis):
+    def __init__(self,input_model,input_parameters,sample):
+        self.sample = sample
+        self.mapping = False
+        self.adjoint_parameters_path =input_parameters["adjoint_parameters_path"].GetString()
+        self.design_surface_sub_model_part_name = input_parameters["design_surface_sub_model_part_name"].GetString()
+        super(SimulationScenario,self).__init__(input_model,input_parameters)
+
+    def Finalize(self):
+
+        super().Finalize()
+        aoa = self.project_parameters["processes"]["boundary_conditions_process_list"][0]["Parameters"]["angle_of_attack"].GetDouble()
+        mach = self.project_parameters["processes"]["boundary_conditions_process_list"][0]["Parameters"]["mach_infinity"].GetDouble()
+        self.primal_model_part = self._GetSolver().main_model_part
+
+
+        with open(self.adjoint_parameters_path,'r') as parameter_file:
+            adjoint_parameters = KratosMultiphysics.Parameters( parameter_file.read() )
+        # Create the adjoint solver
+        adjoint_parameters = _CheckParameters(adjoint_parameters)
+        adjoint_model = KratosMultiphysics.Model()
+
+        adjoint_parameters["processes"]["boundary_conditions_process_list"][0]["Parameters"]["mach_infinity"].SetDouble(mach)
+        adjoint_parameters["processes"]["boundary_conditions_process_list"][0]["Parameters"]["angle_of_attack"].SetDouble(aoa)
+        self.adjoint_analysis = potential_flow_analysis.PotentialFlowAnalysis(adjoint_model, adjoint_parameters)
+
+        self.primal_state_variables = [KCPFApp.VELOCITY_POTENTIAL, KCPFApp.AUXILIARY_VELOCITY_POTENTIAL]
+
+        self.adjoint_analysis.Initialize()
+        self.adjoint_model_part = self.adjoint_analysis._GetSolver().main_model_part
+
+        self._SynchronizeAdjointFromPrimal()
+
+        self.response_function = self.adjoint_analysis._GetSolver()._GetResponseFunction()
+        self.response_function.InitializeSolutionStep()
+        # synchronize the modelparts
+        self.adjoint_analysis.RunSolutionLoop()
+
+    """
+    function introducing the stochasticity in the right hand side
+    input:  self: an instance of the class
+    """
+    def ModifyInitialProperties(self):
```

Comment:
Docstrings should be written below the method definition. See [here](https://www.python.org/dev/peps/pep-0257/) for example.
```suggestion
    def ModifyInitialProperties(self):
        """
        Method introducing the stochasticity in the right hand side.
        """
```